File: TNH/TS10_2021/problem_3.cpp
#include <iostream>
using namespace std;


bool is_palin(string s) {
    int n = s.length();
    for (int i = 0; i < n / 2; i++) {
        if (s[i] != s[n - i - 1]) {
            return false;
        }
    }
    return true;
}


int main() {
    string s, sub_s;
    cin >> s;

    string result = "";
    int n = s.length();
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - i; j++) {
            sub_s = s.substr(j, i + 1);
            if (is_palin(sub_s)) {
                result = sub_s;
            }
        }
    }

    cout << result;
    return 0;
}

File: algorithms/TNH/TS10_2021/problem_3.py
s = input()


def is_palin(s):
    n = len(s)
    for i in range(n // 2):
        if s[i] != s[n - i - 1]:
            return False
    return True


n = len(s)
result = ''
for i in range(n):
    for j in range(n - i):
        sub_s = s[j: j + i + 1]
        if is_palin(sub_s):
            result = sub_s

print(result)
